### DISCUSSION TITLE
v2.2.6 晴空万里，春日繁盛

### DISCUSSION THREAD

**Discussion by qwqcode** [ArtalkJS/Artalk#114] [General]
![image](https://user-images.githubusercontent.com/22412567/164232336-f867ce8b-2d0f-486e-8e63-940eb39b0970.png)

更新 v2.1.1 ~ v2.2.6

## 更新

- 图片上传
- 评论排序
- 评论置顶
- 滑动验证码
- 页面访问量计数
- 样式表细节优化
- 相对路径支持优化
- 前后端版本匹配检测

### 侧边栏

- 找不到站点时自动显示创建表单
- 导入功能 (Artransfer) 优化
- 一键更新页面标题功能

### 表情包

- 表情包动态 / 嵌套加载
- 表情包兼容 OwO 格式

### 配置项

- `conf.server` 不再要求填入 `/api` 路径
- `conf.el` 允许传入 HTMLElement 类型值

### 其他

- 升级 marked 升级到最新版
- 新增 latex 支持 (额外插件引入)
- 新增 highlight js 支持 (额外插件引入)

## 修复

- 管理员 Token 失效重新登录
- Win 11 / macOS BigSur+ 系统版本识别
- 相对链接跳转问题
- 图片验证码尺寸太小问题
- 同时显示两个 Checker Layer 问题
- 需要密码响应信息误判问题
- 侧边栏切换站点翻页问题

## 重构

- 侧边栏独立拆分，并嵌入后端程序
- monorepo 重新组织前端项目
- yarn 切换使用 pnpm

### 图片上传 + 滑动验证码

![图片上传](https://user-images.githubusercontent.com/22412567/164231343-71bf2282-2365-409d-8bdd-481c4e401344.gif)

[Latex](https://user-images.githubusercontent.com/22412567/164231231-f5c5aacb-1d80-41c5-b46f-5ceb3ac3aede.png) /  [Geetest](https://user-images.githubusercontent.com/22412567/164231405-b85c1f1e-df4a-4b02-bea8-df8655ec2131.gif) / [粘贴上传图片](https://user-images.githubusercontent.com/22412567/164231444-8e2c0225-1cc1-4ab7-871b-9bfff3797f4a.gif) / [后端 CLI 一键更新](https://user-images.githubusercontent.com/22412567/164231502-69f95b45-3df3-420e-8408-3a595ba0f056.gif) / [标题一键更新](https://user-images.githubusercontent.com/22412567/164231609-44eb2819-5c1a-43b0-b844-d40a965f1967.gif) / [站点自动创建](https://user-images.githubusercontent.com/22412567/164231703-f1f08e65-a634-4c36-ac27-76ea421d348c.gif)

## 文件体积

36.04 kB gzipped

<img width="353" alt="gzipped" src="https://user-images.githubusercontent.com/22412567/164230902-07f9a1e3-049d-4b97-8208-f72493dd1803.png">

<hr /><em>This discussion was created from the release <a href='https://github.com/ArtalkJS/Artalk/releases/tag/v2.2.6'>v2.2.6 晴空万里，春日繁盛 </a>.</em>

**Comment by inkss** [comment]
最近发版速度惊人
